reana/reana_dev/git.py

```diff
@@ -84,6 +87,50 @@ def get_current_commit(srcdir):
     )
 
 
+def git_is_current_version_tagged(component):
+    """Determine whether the current version in source code is present as a git tag."""
+    tags_history = run_command(
+        "git tag --list", component, display=False, return_output=True
+    )
+    current_version = get_current_component_version_from_source_files(component)
+    return current_version in tags_history if tags_history else False
+
+
+def git_create_release_commit(component):
+    """Create a release commit for the given component."""
+    if "release:" in get_current_commit(get_srcdir(component)):
+        display_message("Nothing to do, last commit is a release commit.", component)
+        return False
+
+    current_version = get_current_component_version_from_source_files(component)
+    if not current_version:
+        display_message(
+            f"Version cannot be autodiscovered from source files.", component
+        )
+        sys.exit(1)
+    elif not git_is_current_version_tagged(component):
+        display_message(
+            f"Current version ({current_version}) "
+            "not present as a git tag, plese release it and add a tag.",
```

Comment:
Hmm. I think this check would prevent the command from running on many of our repositories, because we don't have git tags for every YYYYMMDD version in the past, for example:

```console
$ reana-dev git-create-release-commit -c r-client
[2020-08-11T19:31:40] reana-client: Current version (0.7.0.dev20200706) not present as a git tag, plese release it and add a tag.
```

Moreover, for new components, there will be no known git tags, so we should perhaps introduce a CLI option to make the check optional? E.g. `--no-tag-check`? Or even better `--tag X.Y.Z` which, when the option is passed, it would not auto-discover anything but used the passed value? Would be used for "first tags" releases...

P.S. Note the "plese" typo.